### PR TITLE
Add the ability to actually use OpenSSL

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
Before, it was not possible to enable OpenSSL support in libdrizzle-redux since the configure script didn't support enabling it (though it required openssl to be installed).

Expected results (and performed tests, on linux):
- `./configure`
- `./configure --enable-ssl[=yes]`
- `./configure --disable-ssl` (or `./configure --enable-ssl=no`)

Followed by `ldd libdrizzle/.libs/libdrizzle-redux.so`

In the first and second case, I get:
```
	linux-vdso.so.1 =>  (0x00007fffc69e3000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f235b7f1000)
	libssl.so.1.0.0 => /lib/x86_64-linux-gnu/libssl.so.1.0.0 (0x00007f235b592000)
	libcrypto.so.1.0.0 => /lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007f235b1b6000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f235aea4000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f235ab9e000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f235a7d9000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f235a5c2000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f235a3be000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f235bc25000)
```

Before this P.R., and with the third version, I get:
```
	linux-vdso.so.1 =>  (0x00007ffecefbd000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f51f80c1000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f51f7daf000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f51f7aa9000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f51f76e4000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f51f74cd000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f51f84f4000)
```

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -21,6 +21,32 @@ AC_CANONICAL_BUILD
 AC_ARG_PROGRAM
 AC_USE_SYSTEM_EXTENSIONS
 
+# SSL support, for encrypted connections
```
Comment (andreas-bok-sociomantic) [inline configure.ac]: It should go into a separate macro and be included from the m4 dir.

Comment (andreas-bok-sociomantic) [reply configure.ac]: It seems out of line with the format of the `configure.ac` to have it inline

**Review by andreas-bok-sociomantic — CHANGES_REQUESTED** [review]

**Comment by andreas-bok-sociomantic** [comment]
Checked the build failure on travis 

```sh
libdrizzle/conn.cc:1261:80: error: implicit conversion loses integer precision:
      'size_t' (aka 'unsigned long') to 'int' [-Werror,-Wshorten-64-to-32]
  ...SSL_read(con->ssl, (char*)con->buffer_ptr + con->buffer_size, available_buffer);
     ~~~~~~~~                                                      ^~~~~~~~~~~~~~~~
libdrizzle/conn.cc:1384:61: error: implicit conversion loses integer precision:
      'size_t' (aka 'unsigned long') to 'int' [-Werror,-Wshorten-64-to-32]
      write_size= SSL_write(con->ssl, con->buffer_ptr, con->buffer_size);
```
I have a docker image set up for cross compilation of the project. So I will fix that error

**Comment by andreas-bok-sociomantic** [comment]
Maybe rephrase second ci msg to
 > OpenSSL: Fix broken enable/disable option in configure script

**Comment by mathias-lang-sociomantic** [comment]
Appeased the Travis gods. Gonna split the rest out on Monday :)

**Comment by mathias-lang-sociomantic** [comment]
Updated

**Review by andreas-bok-sociomantic — APPROVED** [review]

**Comment by andreas-bok-sociomantic** [comment]
LGTM
